src/components/Accordion/Accordion.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { FC, useState } from 'react';
+import { action } from '@storybook/addon-actions';
+import up from '../../asset/icon/expanded-up.svg';
+import down from '../../asset/icon/expanded-down.svg';
+import { classNames } from '../../utils/classNames';
+export type AccordionType = 'default' | 'checkbox';
+export interface IAccordionProps {
+  /** set customized style */
+  className?: string;
+  /** set disabled Accordion */
+  disabled?: boolean;
+  /** set Accordion expanded */
+  expanded?: boolean;
+  /** set action on title clicked */
+  onChange?: () => void;
+  /**set Accordion title */
+  title?: string;
+  /**set Accordion summary */
+  summary?: string;
+  /**set Accordion expand icon */
+  expandIcon?: string;
+  /**set Accordion retract icon */
+  retractIcon?: string;
+  /**set Accordion type */
+  type?: AccordionType;
+}
+export type patAccordionProps = IAccordionProps;
+export const Accordion: FC<patAccordionProps> = (props) => {
+  const {
+    className,
+    expanded,
+    disabled,
+    title,
+    summary,
+    onChange,
+    type,
+    ...rest
+  } = props;
+  let initialState: boolean = expanded === undefined ? false : expanded;
+  const [show, setShow] = useState<boolean | false>(initialState);
+  const handleDefaultClick = () => {
+    setShow(!show);
+  };
+  const isExpanded: string = show ? 'isExpanded' : '';
+  const isDisabled: string = disabled ? 'isDisabled' : '';
+  const isControlExpanded: string = props.expanded ? 'isExpanded' : '';
+
+  let styleClasses = classNames('accordion', {
+    isDisabled: !!isDisabled,
+  });
+  if (className) {
+    styleClasses += ' ' + className;
+  }
+
+  let accordion;
+  if (props.onChange === undefined) {
```

Comment:
I believe you could simply just use a conditional statement inside the onClick, for example
```
onClick={props.onChange ? props.onChange : handleDefaultClick}
```
so that you dont need to have two long code block